src/Fhir.Anonymizer.Core/Utility/ReferenceUtility.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Hl7.Fhir.Model;
+
+namespace Fhir.Anonymizer.Core.Utility
+{
+    public class ReferenceUtility
+    {
+        private const string InternalReferencePrefix = "#";
+        private static readonly List<Regex> _literalReferenceRegexes = new List<Regex>
+        {
+            // Regex for absolute or relative url reference, https://www.hl7.org/fhir/references.html#literal
+            new Regex(@"^((http|https)://([A-Za-z0-9\\\/\.\:\%\$])*)?("
```

Comment:
Looks like supported resources list here is hard code. Do we need check validation here? Looks like we only need to do id match here, do not need to valid resource format?

Reply:
The list is hard coded in fhir-net-api library and is also used to encode fhir urls https://github.com/FirelyTeam/fhir-net-api/blob/add8073b8d096f66cde13f54c940048ba4363e0a/src/Hl7.Fhir.Core/Rest/HttpUtil.cs. I think it's fine to trust the source here. We need to validate resource names as the common pattern in a FHIR url is "Resourcename/resourceid" which ensures we extract the right resource id.

Reply:
Is this resource list keeps same between FHIR lib versions? I'm afraid validate resource here would introduce unnecessary complexity. We have validate input/output part, can we put validate logic in particular module?
Could we just Change to use [A-Za-z]+ instead of (ResourceList)?

Reply:
No. It's kept seperately for different Hl7 package versions like r5/r4/stu3. We should use accurate regex to capture resource ids from a reference https://www.hl7.org/fhir/references.html#literal